site/content/en/docs/contrib/building/binaries.md

```diff
@@ -49,6 +54,18 @@ Start the cluster using your built minikube with:
 ./out/minikube start
 ```
 
+## Unit Test and lint
+```shell
+make test
+```
+
+## clean and go mod tidy
+```shell
+make clean
+make gomodtidy
```

Comment:
go mod tidy is not needed when cleaning up. It is needed when adding or removing dependencies.

Make clean is rarely needed (if the build system is correct). go mod tidy should be used before pushing changes to make sure the changes dependencies are updated. Since it is usually not needed running it manually for every push or even for every pr is waste of time. I think the CI should reject PRs that cause requires go mod tidy and users can fix the PR by running go mod tidy.

This is the way we use in ramen:
https://github.com/RamenDR/ramen/blob/main/.github/workflows/ramen.yaml#L56

We combine go mod tidy checks with checking that generated files are not modified which is not great, but the good thing is that PRs cannot be merged with untidy source, and authors can fix this easily.

Reply:
sounds good, we have a lint job on every PR, I think we can add soemthing to it to cause it to Fail if go mod changes in the PR